### ISSUE TITLE
Is the index_add_ function differentiable?

### ISSUE THREAD

**Issue by lizhaohu** [pytorch/pytorch#111734]
### 🚀 The feature, motivation and pitch

        verts_normals = torch.zeros_like(cornea_vertex)
        vertices_faces = cornea_vertex[face_index]

        faces_normals = torch.cross(
            vertices_faces[:, 2] - vertices_faces[:, 1],
            vertices_faces[:, 0] - vertices_faces[:, 1],
            dim=-1,
        )
        unit_faces_normals = safe_normalize(faces_normals)
        verts_normals.index_add_(0, face_index[:, 0], unit_faces_normals)
        verts_normals.index_add_(0, face_index[:, 1], unit_faces_normals)
        verts_normals.index_add_(0, face_index[:, 2], unit_faces_normals)
